### PR TITLE
Add Keras Website Documentation for Layers I

### PR DESCRIPTION
**Pull request by Quincy2014**
## What changes were proposed in this pull request?

Add keras website documentation for dropout, core, pooling, recurrent, convolutional, embedding, normalization and advancedActivations layers.

## How was this patch tested?

Jenkins

### CODE REVIEW

File: docs/docs/KerasStyleAPIGuide/Layers/core.md [hunk 1]
```diff
@@ -86,4 +86,472 @@ Output is
...
+```python
+[[0.23566997 0.8936884  0.21225493 0.08271606 0.5655303  0.9498016 0.27770337 0.66696006]
+ [0.45052916 0.23468265 0.8197383  0.32676312 0.8063288  0.7235357 0.92546755 0.08359752]]
+```
+
+---
+## **Reshape**
+Reshapes an output to a certain shape.
+
+Supports shape inference by allowing one -1 in the target shape. For example, if inputShape = Shape(2, 3, 4), targetShape = Array(3, -1), then outputShape will be Shape(3, 8).
```
Comment (hkvision) [inline docs/docs/KerasStyleAPIGuide/Layers/core.md]: Modify

File: docs/docs/KerasStyleAPIGuide/Layers/core.md [hunk 2]
```diff
@@ -86,4 +86,472 @@ Output is
...
+
+[com.intel.analytics.bigdl.tensor.DenseTensor of size 2x2x3x4]
+```
+Output is:
+```scala
+output: com.intel.analytics.bigdl.nn.abstractnn.Activity =
+-0.33993733	0.6321729	-0.8098835	-0.59065235	-1.5887159	0.715828	-0.85611266	1.0672033	1.2596635	-1.237745	-0.12619138	0.4679131	-0.6356864	-0.2645429	-0.6001224	-0.97786295
+-0.35982707	0.8466941	0.91678447	0.8477781	-0.9805214	0.19218816	0.77162135	0.087835714	0.073235944	-0.86371166	0.16353804	0.052104328	1.6936276	-1.2721802	1.0948435	-0.99680495
+-0.30766153	1.175706	-0.07466983	0.13588612	-1.0921284	-0.55233824	1.8577499	-0.20041502	0.27287027	-0.22101387	0.11674305	-1.1666387	-0.2115776	-0.38883215	0.6829677	-0.40729326
+[com.intel.analytics.bigdl.tensor.DenseTensor of size 3x16]
```
Comment (hkvision) [inline docs/docs/KerasStyleAPIGuide/Layers/core.md]: This is not correct. Rerun the program to verify this.

File: docs/docs/KerasStyleAPIGuide/Layers/core.md [hunk 3]
```diff
@@ -86,4 +86,472 @@ Output is
...
+Permute(dims, inputShape = null)
+```
+**Python:**
+```python
+Permute(dims, input_shape=None)
+```
+
+**Parameters:**
+
+* `dims`: Int array. Permutation pattern, does not include the samples dimension. Indexing starts at 1.
```
Comment (hkvision) [inline docs/docs/KerasStyleAPIGuide/Layers/core.md]: Python is a tuple but not array?

File: docs/docs/KerasStyleAPIGuide/Layers/core.md [hunk 4]
```diff
@@ -86,4 +86,472 @@ Output is
...
+* `inputShape`: Only need to specify this argument when you use this layer as the first layer of a model. For Scala API, it should be a [`Shape`](../keras-api-scala/#shape) object. For Python API, it should be a shape tuple. Batch dimension should be excluded.
+
+**Scala example:**
+```scala
+import com.intel.analytics.bigdl.nn.keras.{Sequential, RepeatVector}
+import com.intel.analytics.bigdl.utils.Shape
+import com.intel.analytics.bigdl.tensor._
+
+val model = Sequential[Float]()
+model.add(RepeatVector(4, inputShape = Shape(3)))
```
Comment (hkvision) [inline docs/docs/KerasStyleAPIGuide/Layers/core.md]: Python and Scala use same argument.

File: docs/docs/KerasStyleAPIGuide/Layers/core.md [hunk 5]
```diff
@@ -86,4 +86,472 @@ Output is
...
+
+---
+## **Merge**
+Used to merge a list of tensors into a single tensor, following some merge mode.
+
+Merge must have at least two input layers.
+
+**Scala:**
+```scala
+Merge(layers =  null, mode = "sum", concatAxis = -1, inputShape = null)
```
Comment (hkvision) [inline docs/docs/KerasStyleAPIGuide/Layers/core.md]: remove redundant space

File: docs/docs/KerasStyleAPIGuide/Layers/core.md [hunk 6]
```diff
@@ -86,4 +86,472 @@ Output is
...
+**Python:**
+```python
+Merge(layers=None, mode="sum", concat_axis=-1, input_shape=None)
+```
+
+**Parameters:**
+
+* `layers`: A list of layer instances. Must be more than one layer.
+* `mode`: Merge mode. String, must be one of: 'sum', 'mul', 'concat', 'ave', 'cos', 'dot', 'max'. Default is 'sum'.
+* `concatAxis`: Integer, axis to use in mode concat. Only specify this when mode is 'concat'. Default is -1, meaning the last axis of the input.
```
Comment (hkvision) [inline docs/docs/KerasStyleAPIGuide/Layers/core.md]: Modify according to the latest.

File: docs/docs/KerasStyleAPIGuide/Layers/core.md [hunk 7]
```diff
@@ -86,4 +86,472 @@ Output is
...
+* `inputShape`: Only need to specify this argument when you use this layer as the first layer of a model. For Scala API, it should be a [`Shape`](../keras-api-scala/#shape) object. For Python API, it should be a shape tuple. Batch dimension should be excluded.
+
+**Scala example:**
+```scala
+import com.intel.analytics.bigdl.nn.keras.{Sequential, Merge}
+import com.intel.analytics.bigdl.utils.Shape
+import com.intel.analytics.bigdl.tensor._
+
+val model = Sequential[Float]()
+model.add(Merge(4, inputShape = Shape(3)))
```
Comment (hkvision) [inline docs/docs/KerasStyleAPIGuide/Layers/core.md]: Refer to MergeSpec. import T.

File: docs/docs/KerasStyleAPIGuide/Layers/advancedActivations.md [hunk 8]
```diff
@@ -0,0 +1,334 @@
+---
+## **ELU**
+Exponential Linear Unit.
+
+It follows: f(x) =  alpha * (exp(x) - 1.) for x < 0, f(x) = x for x >= 0.
+
+The input of this layer should be 2D.
```
Comment (hkvision) [inline docs/docs/KerasStyleAPIGuide/Layers/advancedActivations.md]: require 2D input for ELU?

File: docs/docs/KerasStyleAPIGuide/Layers/advancedActivations.md [hunk 9]
```diff
@@ -0,0 +1,334 @@
...
+* `inputShape`: Only need to specify this argument when you use this layer as the first layer of a model. For Scala API, it should be a [`Shape`](../keras-api-scala/#shape) object. For Python API, it should be a shape tuple. Batch dimension should be excluded.
+
+**Scala example:**
+```scala
+import com.intel.analytics.bigdl.nn.keras.{Sequential, ELU}
+import com.intel.analytics.bigdl.utils.Shape
+import com.intel.analytics.bigdl.tensor._
+
+val model = Sequential[Float]()
+model.add(ELU(1.2, inputShape = Shape(3)))
```
Comment (hkvision) [inline docs/docs/KerasStyleAPIGuide/Layers/advancedActivations.md]: Use default value 1.0

File: docs/docs/KerasStyleAPIGuide/Layers/advancedActivations.md [hunk 10]
```diff
@@ -0,0 +1,332 @@
...
+```
+
+**Python example:**
+```python
+import numpy as np
+from bigdl.nn.keras.topology import Sequential
+from bigdl.nn.keras.layer import ELU
+
+model = Sequential()
+model.add(ELU(1.2, input_shape=(3, )))
```
Comment (hkvision) [inline docs/docs/KerasStyleAPIGuide/Layers/advancedActivations.md]: Change python parameter to 1.0 as well.....
Always scala parameters = python parameters

File: docs/docs/KerasStyleAPIGuide/Layers/advancedActivations.md [hunk 11]
```diff
@@ -0,0 +1,332 @@
...
+
+---
+## **LeakyReLU**
+Leaky version of a Rectified Linear Unit.
+
+It allows a small gradient when the unit is not active: f(x) = alpha * x for x < 0, f(x) = x for x >= 0.
+
+**Scala:**
+```scala
+LeakyReLU(alpha = 0.01, inputShape = null)
```
Comment (hkvision) [inline docs/docs/KerasStyleAPIGuide/Layers/advancedActivations.md]: use default value 0.3.. also for python

Comment (Quincy2014) [reply docs/docs/KerasStyleAPIGuide/Layers/advancedActivations.md]: The default value is 0.01?

Comment (hkvision) [reply docs/docs/KerasStyleAPIGuide/Layers/advancedActivations.md]: Default value should refer to Keras is 0.3 please double check keras doc.

Comment (hkvision) [reply docs/docs/KerasStyleAPIGuide/Layers/advancedActivations.md]: Update and rerun the code to get results.

File: docs/docs/KerasStyleAPIGuide/Layers/advancedActivations.md [hunk 12]
```diff
@@ -0,0 +1,296 @@
...
+input: com.intel.analytics.bigdl.tensor.Tensor[Float] =
+0.8846715	-0.5720033	-0.8220917
+-0.51755846	1.099684	2.6011446
+[com.intel.analytics.bigdl.tensor.DenseTensor of size 2x3]
+```
+Output is:
+```scala
+output: com.intel.analytics.bigdl.nn.abstractnn.Activity =
+0.8846715	    -0.005720033	-0.008220917
+-0.0051755845	1.099684	    2.6011446
```
Comment (hkvision) [inline docs/docs/KerasStyleAPIGuide/Layers/advancedActivations.md]: Format?

File: docs/docs/KerasStyleAPIGuide/Layers/advancedActivations.md [hunk 13]
```diff
@@ -0,0 +1,296 @@
...
+
+---
+## **SReLU**
+S-shaped Rectified Linear Unit.
+
+It follows: f(x) = t^r + a^r(x - t^r) for x >= t^r, f(x) = x for t^r > x > t^l, f(x) = t^l + a^l(x - t^l) for x <= t^l.
+
+**Scala:**
+```scala
+SReLU(tLeftInit = Zeros, aLeftInit = Xavier, tRightInit = Xavier, aRightInit = Ones, sharedAxes = null, inputShape = null)
```
Comment (hkvision) [inline docs/docs/KerasStyleAPIGuide/Layers/advancedActivations.md]: Refer to Object apply method, tLeftInit something should also be string.

File: docs/docs/KerasStyleAPIGuide/Layers/advancedActivations.md [hunk 14]
```diff
@@ -0,0 +1,296 @@
...
+```
+**Python:**
+```python
+SReLU(t_left_init='zero', a_left_init='glorot_uniform', t_right_init='glorot_uniform', a_right_init='one', shared_axes=None, input_shape=None)
+```
+
+**Parameters:**
+
+* `tLeftInit`: Initialization function for the left part intercept. Default is Zeros. You can also pass in corresponding string representations such as 'zero' or 'normal', etc. for simple init methods in the factory method.
```
Comment (hkvision) [inline docs/docs/KerasStyleAPIGuide/Layers/advancedActivations.md]: Refer to python doc.....

File: docs/docs/KerasStyleAPIGuide/Layers/advancedActivations.md [hunk 15]
```diff
@@ -0,0 +1,296 @@
...
+Input is:
+```scala
+input: com.intel.analytics.bigdl.tensor.Tensor[Float] =
+(1,.,.) =
+0.5599429	0.22811626	-0.027771426
+-0.56582874	1.9261217	1.2686813
+
+(2,.,.) =
+0.7538568	0.8725621	0.19803657
+0.49057	    0.0537252	0.8684544
```
Comment (hkvision) [inline docs/docs/KerasStyleAPIGuide/Layers/advancedActivations.md]: format?

File: docs/docs/KerasStyleAPIGuide/Layers/advancedActivations.md [hunk 16]
```diff
@@ -0,0 +1,296 @@
...
+[[0.91854565 0.58317415 0.33089385]
+ [0.82472184 0.70572913 0.32803604]]
+```
+Output is
+```python
+[[0.0   0.0     0.0]
+ [0.0   0.0     0.0]]
+```
+
+---
```
Comment (hkvision) [inline docs/docs/KerasStyleAPIGuide/Layers/advancedActivations.md]: Need this line?

File: docs/docs/KerasStyleAPIGuide/Layers/advancedActivations.md [hunk 17]
```diff
@@ -0,0 +1,296 @@
...
+output = model.forward(input)
+```
+Input is:
+```python
+[[0.91854565 0.58317415 0.33089385]
+ [0.82472184 0.70572913 0.32803604]]
+```
+Output is
+```python
+[[0.0   0.0     0.0]
```
Comment (hkvision) [inline docs/docs/KerasStyleAPIGuide/Layers/advancedActivations.md]: space?

File: docs/docs/KerasStyleAPIGuide/Layers/advancedActivations.md [hunk 18]
```diff
@@ -0,0 +1,296 @@
+---
```
Comment (hkvision) [inline docs/docs/KerasStyleAPIGuide/Layers/advancedActivations.md]: Remove the first line for every file.

File: docs/docs/KerasStyleAPIGuide/Layers/convolutional.md [hunk 19]
```diff
@@ -0,0 +1,887 @@
...
+## **Convolution2D**
+Applies a 2D convolution over an input image composed of several input planes.
+
+You can also use Conv2D as an alias of this layer.
+
+The input of this layer should be 4D.
+
+**Scala:**
+```scala
+Convolution2D(nbFilter, nbRow, nbCol, init = Xavier, activation = null, borderMode = "valid", subsample = Array(1, 1), dimOrdering = DataFormat.NCHW, wRegularizer = null, bRegularizer = null, bias = true, inputShape = null)
```
Comment (hkvision) [inline docs/docs/KerasStyleAPIGuide/Layers/convolutional.md]: init = "glorot_uniform"

File: docs/mkdocs.yml [hunk 20]
```diff
@@ -98,6 +98,14 @@ pages:
     - Activation: KerasStyleAPIGuide/Layers/activation.md
     - Initialization: KerasStyleAPIGuide/Layers/initialization.md
     - Core Layers: KerasStyleAPIGuide/Layers/core.md
+    - Pooling Layers: KerasStyleAPIGuide/Layers/pooling.md
+    - Recurrent Layers: KerasStyleAPIGuide/Layers/recurrent.md
+    - Dropout Layers: KerasStyleAPIGuide/Layers/dropout.md
+    - Convolutional Layers: KerasStyleAPIGuide/Layers/convolutional.md
+    - Embedding Layers: KerasStyleAPIGuide/Layers/embedding.md
+    - Normalization Layers: KerasStyleAPIGuide/Layers/normalization.md
+    - Advanced Activations Layers: KerasStyleAPIGuide/Layers/advancedActivations.md
```
Comment (hkvision) [inline docs/mkdocs.yml]: rename to advanced-activation.md

File: docs/mkdocs.yml [hunk 21]
```diff
@@ -98,6 +98,14 @@ pages:
     - Activation: KerasStyleAPIGuide/Layers/activation.md
     - Initialization: KerasStyleAPIGuide/Layers/initialization.md
     - Core Layers: KerasStyleAPIGuide/Layers/core.md
+    - Pooling Layers: KerasStyleAPIGuide/Layers/pooling.md
+    - Recurrent Layers: KerasStyleAPIGuide/Layers/recurrent.md
+    - Dropout Layers: KerasStyleAPIGuide/Layers/dropout.md
+    - Convolutional Layers: KerasStyleAPIGuide/Layers/convolutional.md
```
Comment (hkvision) [inline docs/mkdocs.yml]: reorder: convolutional, pooling, recurrent, normalization, embedding, dropout, advanced activation

File: docs/docs/KerasStyleAPIGuide/Layers/wrappers.md [hunk 22]
```diff
@@ -0,0 +1,178 @@
...
+[[[ 0.11651072  0.07040063  0.53200144 -0.37872505]
+  [ 0.03238479  0.15081021  0.55530167 -0.3390156 ]
+  [ 0.18388109  0.02891854  0.5591757  -0.28601688]]
+
+ [[-0.17779878 -0.02685877  0.244566   -0.34734237]
+  [-0.17816684  0.077871    0.3195565  -0.40989208]
+  [-0.13442594  0.08941883  0.3418655  -0.29824993]]]
+```
+
+---
```
Comment (hkvision) [inline docs/docs/KerasStyleAPIGuide/Layers/wrappers.md]: remove

File: docs/docs/KerasStyleAPIGuide/Layers/recurrent.md [hunk 23]
```diff
@@ -0,0 +1,335 @@
...
+[[0.5762107  0.45679288 0.00370956]
+ [0.24133312 0.38104653 0.05249192]]
+```
+Output is:
+```python
+[[0.5762107  0.4567929  0.00370956]
+ [0.24133313 0.38104653 0.05249191]]
+```
+
+---
```
Comment (hkvision) [inline docs/docs/KerasStyleAPIGuide/Layers/recurrent.md]: remove

**Comment by Quincy2014** [comment]
Jenkins passed. http://172.168.2.101:8080/job/BigDL-PR-Validation-v3/415/
